### DISCUSSION TITLE
Memory management for large batch sizes

### DISCUSSION THREAD

**Discussion by pj-ml** [sgl-project/sglang#187] [Q&A] [answered]
Does SGLang automatically manage memory when a large batch (e.g. 100 batch size with avg. token per sequence 6,000) is submitted for processing?

I have been getting OOM errors when increasing my batch sizes, which is surprising because I thought the memory management was automatic.

**Comment by hnyls2002** [comment, marked as answer]
@pj-ml, could you try reducing the argument `--mem-fraction-static`? When the batch size increases, the backend may require more GPU memory for temporary usage.

https://github.com/sgl-project/sglang/?tab=readme-ov-file#additional-arguments
